site.ts

```diff
@@ -329,22 +356,72 @@ export function inject({ config, posthog }) {
+        </div>
+
+        </div>
+        `
+        const formElement = Object.assign(document.createElement('form'), {
+            className: `survey-${survey.id}-form`,
+            innerHTML: form,
+            onsubmit: (e) => {
+                e.preventDefault()
+                const selectedChoices = singleOrMultiSelect === 'multiple_single' ? e.target.querySelector('input[type=radio]:checked').value : [...e.target.querySelectorAll('input[type=checkbox]:checked')].map((choice) => choice.value)
+                posthog.capture('survey sent', {
+                    $survey_name: survey.name,
+                    $survey_id: survey.id,
+                    $survey_question: survey.question,
+                    $survey_response: selectedChoices,
+                    sessionRecordingUrl: posthog.get_session_replay_url(),
+                })
+                closeSurveyPopup(survey.id, formElement)
+            }
+        })
+        return formElement
+    }
+
     const callSurveys = (posthog, forceReload = false) => {
         posthog?.getActiveMatchingSurveys((surveys) => {
-            const nonAPISurveys = surveys.filter(survey => survey.type !== 'api')
-            nonAPISurveys.forEach((survey) => {
-                if (document.querySelectorAll("div[class^='PostHogSurvey']").length === 0) {
-                    if (!localStorage.getItem(`seenSurvey_${survey.id}`)) {
-                        const shadow = createShadow(style(survey.id, survey?.appearance), survey.id)
-                        let surveyPopup
-                        const surveyQuestionType = survey.questions[0].type
-                        if (surveyQuestionType === 'rating') {
-                            surveyPopup = createRatingsPopup(survey)
-                        } else if (surveyQuestionType === 'open' || surveyQuestionType === 'link') {
-                            surveyPopup = createSurveyPopup(survey)
-                        }
-                        addCancelListeners(surveyPopup, survey.id, survey.name)
-                        shadow.appendChild(surveyPopup)
+        const nonAPISurveys = surveys.filter(survey => survey.type !== 'api')
+        nonAPISurveys.forEach((survey) => {
+            if (document.querySelectorAll("div[class^='PostHogSurvey']").length === 0) {
+                if (!localStorage.getItem(`seenSurvey_${survey.id}`)) {
+                    const shadow = createShadow(style(survey.id, survey?.appearance), survey.id)
+                    let surveyPopup
+                    const surveyQuestionType = survey.questions[0].type
+                    if (surveyQuestionType === 'rating') {
+                        surveyPopup = createRatingsPopup(survey)
+                    } else if (surveyQuestionType === 'open' || surveyQuestionType === 'link') {
+                        surveyPopup = createSurveyPopup(survey)
+                    } else if (surveyQuestionType === 'multiple_single' || surveyQuestionType === 'multiple_multi') {
```

Comment:
these type names seem a bit awkward, since they're not internal only, and anyone building a survey will use them.

Suggestion, rename them to: `single_choice` and `multiple_choice`, since it's clear that single choice doesn't mean there's only one choice, but only choice you can select.